src/components/DataSampleGridView.tsx

```diff
@@ -56,37 +77,80 @@ const DataSamplesGridView = () => {
 
   return (
     <>
-      <RecordingDialog
-        gestureId={selectedGesture.ID}
-        isOpen={isOpen}
-        onClose={onClose}
-        actionName={selectedGesture.name}
+      <ConnectToRecordDialog
+        isOpen={connectToRecordDialogDisclosure.isOpen}
+        onClose={connectToRecordDialogDisclosure.onClose}
       />
+      {selectedGesture && (
+        <RecordingDialog
+          gestureId={selectedGesture.ID}
+          isOpen={isOpen}
+          onClose={onClose}
+          actionName={selectedGesture.name}
+        />
+      )}
       <HeadingGrid
         position="sticky"
         top={0}
         {...gridCommonProps}
         headings={headings}
       />
-      <Grid
-        {...gridCommonProps}
-        alignItems="start"
-        autoRows="max-content"
-        overflow="auto"
-        flexGrow={1}
-        h={0}
-      >
-        {gestures.map((g, idx) => (
-          <DataSampleGridRow
-            key={g.ID}
-            gesture={g}
-            selected={selectedGesture.ID === g.ID}
-            onSelectRow={() => setSelectedGestureIdx(idx)}
-            startRecording={onOpen}
-            showWalkThrough={showWalkThrough}
-          />
-        ))}
-      </Grid>
+      {gestures.length === 0 ? (
+        <VStack
+          gap={5}
+          flexGrow={1}
+          alignItems="center"
+          justifyContent="center"
+        >
+          <LoadProjectInput ref={loadProjectInputRef} accept=".json" />
+          <Text fontSize="lg">
+            <FormattedMessage id="no-data-samples" />
+          </Text>
+          {!isConnected && (
+            <Text fontSize="lg">
+              <Button
+                fontSize="lg"
+                color="brand.600"
+                variant="link"
+                onClick={actions.startConnect}
+              >
+                Connect a micro:bit
```

Comment:
We need to extract this whole thing with link1/link2 tags.

Reply:
[320b30a](https://github.com/microbit-foundation/ml-trainer/pull/331/commits/320b30abde8c942526e796505d5f3a091411425d)